src/migrator/sync.ts

```diff
@@ -0,0 +1,177 @@
+import {
+	modelText,
+	MigrationTuple,
+	MigrationError,
+	defaultMigrationCategory,
+	checkModelAlreadyExists,
+	setExecutedMigrations,
+	getExecutedMigrations,
+	lockMigrations,
+	MigrationCategories,
+	RunnableMigrations,
+} from './utils';
+import type { Tx } from '../database-layer/db';
+import type { Config, Model } from '../config-loader/config-loader';
+
+import * as _ from 'lodash';
+import * as sbvrUtils from '../sbvr-api/sbvr-utils';
+
+type ApiRootModel = Model & { apiRoot: string };
+
+export const postRun = async (tx: Tx, model: ApiRootModel): Promise<void> => {
+	const { initSql } = model;
+	if (initSql == null) {
+		return;
+	}
+
+	const modelName = model.apiRoot;
+
+	const exists = await checkModelAlreadyExists(tx, modelName);
+	if (!exists) {
+		(sbvrUtils.api.migrations?.logger.info ?? console.info)(
+			'First time executing, running init script',
+		);
+
+		await lockMigrations(tx, modelName, async () => {
+			try {
+				await tx.executeSql(initSql);
+			} catch (err) {
+				(sbvrUtils.api.migrations?.logger.error ?? console.error)(
+					`postRun locked sql execution error ${err} `,
+				);
+				throw new MigrationError(err);
+			}
+		});
+	}
+};
+
+export const run = async (tx: Tx, model: ApiRootModel): Promise<void> => {
+	const { migrations } = model;
+	if (
+		migrations == null ||
+		_.isEmpty(migrations) ||
+		migrations[defaultMigrationCategory] === undefined
+	) {
+		return;
+	}
+
+	let runMigrations: RunnableMigrations = {};
+	for (const [key, value] of Object.entries(migrations)) {
+		if (key === defaultMigrationCategory && value instanceof Object) {
```

Comment:
What's the reason for instanceof vs the more common typeof?
```suggestion
		if (key === defaultMigrationCategory && typeof value === 'object') {
```

Reply:
Mo specific reasoning. At some point my IDE (VS Code) was complainingabout checking for tyepof <var> === 'object' and I used instanceof. I didn't changed it back.

Reply:
I expect it was because whilst `typeof x === 'object'` is generally more standard `null` is also a type of object